pruning/pruning_test.go

```diff
@@ -83,7 +83,7 @@ func TestPruneSnapshots(t *testing.T) {
 		"simple": {
 			inputs: inputs["s1"],
 			rules: []KeepRule{
-				MustKeepRegex("foo_", false),
+				MustKeepRegex(map[string]bool{}, "foo_", false),
```

Comment:
Please find a way to avoid the clutter with the empty map this test function.
And please add another test function that tests the functionality which you added.

To accomplish the former, I suggest the following steps:
1. you introduce the `KeepCommon` concept to the MustKeep* family of functions.
2. use a default `KeepCommon` for all MustKeep calls within this function. That default doesn't have a `filesystems` filter.

Then, for the additional tests, you'll likely just vary the KeepCommon.